### DISCUSSION TITLE
Wazuh - Authentication error. Wrong key or corrupt payload

### DISCUSSION THREAD

**Discussion by rdemars-zcorum** [Security-Onion-Solutions/securityonion#2432] [Unsupported Versions]
Running Security Onion 2.3.21 and getting the follow Wazuh errors on the manager:
2020/12/22 17:58:55 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '004' at '172.26.10.10'.
2020/12/22 17:59:08 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '005' at '10.75.236.109'.
2020/12/22 17:59:27 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '007' at '10.75.236.122'.
2020/12/22 17:59:37 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '006' at '10.75.236.3'.
2020/12/22 17:59:51 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '002' at '172.26.7.4'.
2020/12/22 17:59:57 ossec-remoted: WARNING: (1404): Authentication error. Wrong key or corrupt payload. Message received from agent '003' at '10.75.236.245'.

Versions:
[root@so-manager /]# docker exec -it so-wazuh dpkg -l |grep wazuh
ii  wazuh-api      3.13.1-1     amd64        Wazuh API is an open source RESTf
ii  wazuh-manager  3.13.1-1     amd64        Wazuh helps you to gain security
[root@agentsystem ossec]# rpm -qa |grep wazuh
wazuh-agent-3.13.2-1.x86_64

Have the extracted keys for the agents from so-wazuh-agent-manage with them installed in the /var/ossec/etc/client.keys files on the agents.

**Comment by weslambert** [comment]
Have you moved or copied config from another install, or is this from a stock install with agents added?

**Reply by rdemars-zcorum** [reply]
This is a fresh install of Security Onion with Wazuh and install of Wazuh agent on CentOS 7 servers. Keys are extracted using so-wazuh-agent-manage on the so-manager.

**Reply by weslambert** [reply]
Have you tried just running `manage_agents` and choosing the `I` option from the endpoint(s)?

**Comment by rdemars-zcorum** [comment]
Found that so-wazuh-agent-manage was not exporting the key correctly. Used "docker exec -it so-wazuh cat /var/ossec/etc/client.keys" to find the client keys to use.

**Comment by weslambert** [comment]
`so-wazuh-agent-manage` exports keys to be used with `manage_agents` -- this is a different key that is derived from the Agent ID, Name, IP address, and underlying key in `client.keys`.  As you mentioned, if you are copying directly to client.keys, then you would need to copy the keys from `/nsm/wazuh/etc/client.keys` (which is essentially the same as what you did previously, but avoids having to mess with Docker).  I just wanted to be clear that `so-wazuh-manage-agent` is working as it should.
